pallets/ah-migrator/src/account.rs

```diff
@@ -262,15 +275,111 @@ impl<T: Config> crate::types::AhMigrationCheck for AccountsMigrator<T> {
 	/// the check that data has been correctly migrated to asset hub. It should also contain the
 	/// data previously stored in asset hub, allowing for more complex logical checks on the
 	/// migration outcome.
-	fn post_check(_rc_total_issuance_before: Self::RcPrePayload, _: Self::AhPrePayload) {
+	fn post_check(rc_pre_payload: Self::RcPrePayload, ah_pre_payload: Self::AhPrePayload) {
 		// Check that no failed accounts remain in storage
 		assert!(
 			RcAccounts::<T>::iter().next().is_none(),
 			"Failed accounts should not remain in storage after migration"
 		);
 
-		// TODO: Giuseppe @re-gius
-		//   run post migration sanity checks like:
-		//    - rc_migrated_out == ah_migrated_in - failed accounts
+		let (account_summaries, _) = rc_pre_payload;
+		for (who, summary) in account_summaries {
+			// assert_eq!(summary.migrated_free, <T as Config>::Currency::balance(&who));
+			// assert_eq!(summary.migrated_reserved, <T as
+			// Config>::Currency::reserved_balance(&who));
+			let ah_free_post = <T as Config>::Currency::balance(&who);
+			let ah_reserved_post = <T as Config>::Currency::reserved_balance(&who);
+			let mut ah_free_before = 0;
+			let mut ah_reserved_before = 0;
+			if let Some((ah_reserved_pre, ah_free_pre)) = ah_pre_payload.get(&who) {
```

Comment:
let (ah_free_before, ah_reserved_before) if {} else {}, and no need for mutable. always easier to follow, since I know its not gonna be mutated

Reply:
this is minor